nixos/modules/programs/teck-programmer.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.teck-programmer;
+
+in
+{
+  options.programs.teck-programmer = {
+    enable = mkEnableOption "user access to TECK keyboard firmware";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.teck-programmer;
+      description = "the teck-programmer package";
+    };
```

Comment:
Is there any other option? Do we need this?

Reply:
Potentially: there exists [one in C++](https://github.com/m-ou-se/tek) and [one in Rust](https://github.com/zgtm/tek-programmer-rs).

Reply:
That sounds like a reason to keep the package option around :+1: